library/azure_rm_resource.py

```diff
@@ -0,0 +1,296 @@
+#!/usr/bin/python
+#
+# Copyright (c) 2018 Zim Kalinowski, <zikalino@microsoft.com>
+#
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+from __future__ import absolute_import, division, print_function
+__metaclass__ = type
+
+
+ANSIBLE_METADATA = {'metadata_version': '1.1',
+                    'status': ['preview'],
+                    'supported_by': 'community'}
+
+
+DOCUMENTATION = '''
+---
+module: azure_rm_resource
+version_added: "2.6"
+short_description: Create any Azure resource.
+description:
+  - Create any Azure resource using Azure REST API.
+
+options:
+  url:
+    description:
+      - Azure RM Resource URL.
+  provider:
+    description:
+      - Provider type, should be specified in no URL is given
+  resource_group:
+    description:
+      - Resource group to be used, should be specified if needed and URL is not specified
+  resource_type:
```

Comment:
add choices?

Reply:
there would be too many, maybe around 1000 and they depend on provider, so that would be too complex

Reply:
or at least give user a link on where to find provider list. otherwise, it's hard for user to know what should be a right value for string type option.

Reply:
yes, i think i can add reference to Azure API. Users of this module should be considered advanced, and they won't be able to write task from scratch without referring to REST API, using existing sample or using VSC extension

Reply:
done

Reply:
so no required options for this module?

Reply:
only conditionally required, and conditions may be pretty complex, so this will be validated by underlying rest.